### PR TITLE
docs: add async validator example

### PR DESCRIPTION
**Pull request by loopingz**
I added an example for asynchronous and a sentence to explain they can be either sync or async.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,6 +132,30 @@ const validator = (value) => {
     // Parse the value, modifying it
     return value.replace('aa', 'bb');
 }
+
+const asyncValidator = async (value) => {
```
Comment (satazor) [inline README.md:136]: The example seems a bit too complicated, can we make it simple by just awaiting “someAsyncAssert”, or similar?

Comment (loopingz) [reply README.md:136]: I m up for both approach, I tried to add a remote fetch to make it more 'real' but if you prefer a simpler like:

```
async (value) => {
  await myfunc();
  return value;
}
```


You prefer the last one?

Comment (satazor) [reply README.md:136]: yep!

Comment (loopingz) [reply README.md:136]: Here you are Sir!

**Review by satazor — APPROVED** [review]

**Pull request merged**